CMakeLists.txt

```diff
@@ -379,36 +379,45 @@ if (WITH_GPU)
       ${SYCL_ONLY_SRC_FILES})
   target_link_libraries(general-sycl-offloading PRIVATE SeisSol-common-properties)
 
-  # add GPU dynamic rupture target
-  if (DEVICE_ARCH MATCHES "sm_*")
-    if (CMAKE_CXX_COMPILER_ID MATCHES "NVHPC|PGI")
-      # we assume that hipsycl was compiled with nvhpc compiler collection
-      string(REPLACE sm_ cc NVCPP_ARCH ${DEVICE_ARCH})
-      set(HIPSYCL_TARGETS "cuda-nvcxx:${NVCPP_ARCH}")
+  set(SYCL_CC "hipsycl" CACHE STRING "The SYCL Compiler used for general SYCL offloading")
+  set(SYCL_CC_OPTIONS hipsycl dpcpp)
+  set_property(CACHE SYCL_CC PROPERTY STRINGS ${SYCL_CC_OPTIONS})
+
+  if (SYCL_CC MATCHES "hipsycl")
+    message(STATUS "Using HipSYCL for general SYCL offloading")
+    # add GPU dynamic rupture target
+    # always override HIPSYCL_TARGETS, as we may need to override it when the architecture changes
+    if (DEVICE_ARCH MATCHES "sm_*")
+      if (CMAKE_CXX_COMPILER_ID MATCHES "NVHPC|PGI")
+        set(SYCL_USE_NVHPC_DEFAULT ON)
+      else()
+        set(SYCL_USE_NVHPC_DEFAULT OFF)
+      endif()
+      option(SYCL_USE_NVHPC "For Nvidia GPUs, use nvhpc instead of CUDA/nvcc." ${SYCL_USE_NVHPC_DEFAULT})
```

Comment:
I would suggest to you the following script:

https://github.com/SeisSol/Device/blob/ravil/opensycl/cmake/find-opensycl.py

and the corresponding cmake file

https://github.com/SeisSol/Device/blob/ravil/opensycl/cmake/FindOpenSYCLSettings.cmake

It is from the following PR: https://github.com/SeisSol/Device/pull/21

The idea is that one may want to configure SeisSol with just SYCL and use `hipsycl`+`nvhpc` as a backend.

Currently, the change in this PR need to be applied in three places: DR, Device and `src/sycl.cmake`.

However, the solution that I am proposing will result in something like
```
  find_package(OpenSYCLSettings REQUIRED)
  find_package(OpenSYCL REQUIRED)
  target_link_libraries(SeisSol-device-lib PRIVATE opensycl-settings::device_flags)
```
in three parts of the code.

You can see all required changes here: https://github.com/SeisSol/SeisSol/pull/832